backend/mora/service/manager.py

```diff
@@ -13,20 +13,109 @@
 
 """
 import uuid
+import operator
+import flask
 
 from . import address
 from . import handlers
+from . import orgunit
+from . import facet
+from . import employee
 from .validation import validator
 from .. import common
 from .. import lora
 from .. import mapping
 from .. import util
 
 
-class ManagerRequestHandler(handlers.OrgFunkRequestHandler):
+class ManagerRequestHandler(
+    handlers.OrgFunkRequestHandler,
+    handlers.OrgFunkReadingRequestHandler
+):
+
+    __slots__ = ()
     role_type = 'manager'
     function_key = mapping.MANAGER_KEY
 
+    @classmethod
+    def finder(cls, c, type, objid):
+
+        found = super().finder(c, type, objid)
+        if not found and util.get_args_flag("inherit_manager"):
+            ou = orgunit.get_one_orgunit(
+                c, objid, details=orgunit.UnitDetails.FULL
+            )
+            while not found:
+                upperid = ou[mapping.PARENT][mapping.UUID]
+                ou = orgunit.get_one_orgunit(
+                    c, upperid, details=orgunit.UnitDetails.FULL
+                )
+                found = super().finder(c, type, upperid)
```

Comment:
What happens if we reach the top of the organisation tree without finding a manager?

Reply:
I had nightmares about that during the weekend. 
I have added a check for None on manager before attempting to get the uuid.